docs/question/server.md

```diff
@@ -36,6 +36,18 @@ def generate(data):
     data["correct_answers"]["y"] = 2 * data["params"]["x"]
 ```
 
+Parameters set within `generate` can be used in the `question.html` file by using `{{params.NAME}}`. For example, you can use `data["params"]["x"]` with `{{params.x}}`. You can set parameters in other phases as well, but they may not be defined when the student initially views the question (only after a save or a submission). You can view the [question lifecycle diagram](#question-lifecycle) for more details.
```

Comment:
```suggestion
These values can be used in the `question.html` file with Mustache syntax, e.g. `{{params.NAME}}`. For example, you can use `data["params"]["x"]` with `{{params.x}}`. You can set parameters in some other phases as well, but they may not be defined when the student initially views the question (only after a save or a submission). See the [question lifecycle diagram](#question-lifecycle) for more details.
```

Nit: saves are a subset of submissions. Maybe just "only after a submission"? Or just drop the parenthetical entirely in favor of the link to the question lifecycle.

TBH I don't think we lose much if we drop the note about other phases entirely, I think it feels kind of out of place here. But I'm ok if you want to leave it in.

Reply:
Two reasons:

- It's not only parameters; you could in theory use correct answers too.
- It seemed to be unnecessary to mention generate here. It's already under a big old "Step 1: generate" title, the sentence before the code block mentions "generate" multiple times, the code block itself shows `generate()`. I don't see a need to further reiterate that we're talking about "generate".

Reply:
I completely reworked the language in 302209f5349a4921a887b01de00d449ddd430ae9.